aiida/tools/groups/grouppaths.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Provides functionality for managing large numbers of AiiDA Groups, via label delimitation."""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+from collections import namedtuple
+from functools import total_ordering
+import re
+from typing import Any, Iterable, List, Optional  # pylint: disable=unused-import
+import warnings
+
+from aiida import orm
+from aiida.common.exceptions import NotExistent
+
+__all__ = ('GroupPath', 'InvalidPath')
+
+REGEX_ATTR = re.compile('^[a-zA-Z][\\_a-zA-Z0-9]*$')
+
+
+class InvalidPath(Exception):
+    pass
+
+
+WalkNodeResult = namedtuple('WalkNodeResult', ['group_path', 'node'])
+
+
+@total_ordering
+class GroupPath:
+    """A class to provide label delimited access to groups.
+
+    See tests for usage examples.
+    """
+
+    def __init__(self, path='', type_string=orm.GroupTypeString.USER.value, warn_invalid_child=True):
+        # type: (str, Optional[str], Optional[GroupPath])
+        """Instantiate the class.
+
+        :param path: The initial path of the group.
+        :param type_string: Used to query for and instantiate a ``Group`` with.
+        :param warn_invalid_child: Issue a warning, when iterating children, if a child path is invalid.
+
+        """
+        self._delimiter = '/'
+        if type_string is not None and not isinstance(type_string, str):
+            raise AssertionError('type_string must be None or str: {}'.format(type_string))
+        self._type_string = type_string
+        self._path_string = self._validate_path(path)
+        self._path_list = self._path_string.split(self._delimiter) if path else []
+        self._warn_invalid_child = warn_invalid_child
+
+    def _validate_path(self, path):
+        """Validate the supplied path."""
```

Comment:
You should aready be able to initiate the `GroupPath` with GroupPath(warn_invalid_child=False), to produce the 'ignore' behaviour you describe (but this shouldn't be the default).
Probably though there should be an option on the CLI to use this API feature.

Reply:
But in what case would you want a "broken" list? I mean, as I see it you can either get an already clean list which you can iterate without problems (although might be incomplete, depending on what you aim to do) or you want it to return an error telling you there are groups that can't be parsed this way. The broken list just pushes the error one step further so when I try to use `bad` as a group_path I may get an error saying it doesn't exist, yet the direct resources within this tool will tell me it does (I will only notice if I check for contradictions such that is both virtual and has no children or by using external tools such as manually querying for the group). I see no case where this doesn't just obfuscate the understanding of the situation.

Reply:
I'll have a think on it 😁

Reply:
In 9088575 there is now a `verdi group path ls --no-warn` option which will not omit the warning.

But I'm not implemented the other behaviour, because I think it overcomplicates the situation. For example what if you have `group\\bad` and `group\good`, you're not going to exclude `group` then. So now you need to do some complex look ahead for every child, to decide if they have any valid children of their own and thus whether to exclude them.

I'm pretty user that users that want to use this feature will make all their groups compliant, and the warning is sufficient. But fell free to make a PR of your own after this 😁 if you want to implement it